packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -776,7 +777,12 @@ function importSourceToScope(
 
   let resolved: string | undefined = scope.resolvedImportSources[source]
   if (!resolved) {
-    if (source.startsWith('.')) {
+    if (source.startsWith('..')) {
+      const osSpecificJoinFn = process.platform === 'win32' ? join : joinPaths
```

Comment:
I think `path.join` is already os-specific?

Reply:
Well yes. But thats not about `path.join` itself but about `path.join` VS `path.posix.join` which is coming from `joinPath` function.
On Windows use `path.join` every where else use `path.posix.join` is `path.posix` is available.

Made it so for backward compatibility

So this condition can be unwrapped to:
`(process.platform === 'win32' ? path : path.posix || path).join`

Reply:
TBH I don't understand why `pathJoin` exist cause as you said `path.join` is already OS specific, and `(path.posix || path).join` is always `path.posix.join` as last one is always present. So my assumption is that it's "needed in some cases", so use `path` on windows and fallback to `pathJoin` as it was initially before the fix =).